codec-http/src/main/java/io/netty/handler/codec/http/ServerCookieDecoder.java

```diff
@@ -157,20 +119,52 @@
                     if (i == headerLen) {
                         // NAME (no value till the end of string)
                         newNameEnd = headerLen;
-                        value = null;
+                        newValueStart = newValueEnd = -1;
                         break;
                     }
                 }
             }
 
-            if (!rfc2965Style || (!header.regionMatches(newNameStart, "$Path", 0, "$Path".length()) &&
-                    !header.regionMatches(newNameStart, "$Domain", 0, "$Domain".length()) &&
-                    !header.regionMatches(newNameStart, "$Port", 0, "$Port".length()))) {
+            if (newValueStart == -1) {
+                logger.debug("Skipping cookie with null value");
+                continue;
+            }
+
+            if (rfc2965Style && (header.regionMatches(newNameStart, "$Path", 0, "$Path".length()) ||
+                    header.regionMatches(newNameStart, "$Domain", 0, "$Domain".length()) ||
+                    header.regionMatches(newNameStart, "$Port", 0, "$Port".length()))) {
```

Comment:
Please pre-calculate the length of all of the three, store in a static field and re-use.